docs/overview/architecture.md

```diff
@@ -0,0 +1,245 @@
+8. Vald Agent returns the searching result to the Vald LB Gateway. The searching result includes the UUID, the vector distance, and the vector. The number of the result will be the same as requested.
+9. Vald LB Gateway will aggregate all searching results from all Vald Agents, rank the result by the vector distance, and return the ranked result to the Vald Meta Gateway.
+10. Vald Meta Gateway will forward the searching result to the Vald Meta to resolve the user-defined vector IDs from the internal used UUIDs.
+11. Vald Meta will perform a search for the Vector IDs based on the internal used UUIDs.
+12. Vald Meta will return the Vector IDs to the Vald Meta Gateway.
+13. Vald Meta Gateway will combine the vectors and the vector IDs from the searching result and return to the Vald Filter Gateway.
+14. Vald Filter Gateway will forward the request to the user-defined Vald Egress Filter to filter the final result. For example exclude the specific type of the result from the vector ID.
+15. Vald Egress Filter will return the filtered result to the Vald Filter Gateway.
+16. Vald Filter Gateway will return the final result to the Vald Ingress.
+
+<!-- ### Update -->
+
+<!-- ### Delete -->
+
+## Components
+
+### Vald Filter
+
+Vald Filter is an optional functionality in Vald.
+User can implement the custom filtering logic and integrate with Vald.
+
+Vald Filter provides the following functionalities.
+
+- Custom filter based on request query
+- Custom filter for the searching result
+
+#### Vald Ingress Filter
+
+Vald Ingress Filter filters the incoming request before processing it.
+
+Users can implement custom filtering logic such as changing the vectors or filtering based on user ID.
+
+#### Vald Egress Filter
+
+Vald Egress Filter filters the response before sending it to the user.
+
+This component can reorder the searching result from multiple Vald Agents based on the user-defined ranking.
+
+#### Vald Filter Gateway
+
+Vald Filter Gateway forwards the request to Vald Ingress Filter before processing it and forwards the response to the Vald Egress Filter before returning the searching result to the user.
+
+### Vald Metadata
+
+In Vald, metadata consists of the vector data and the corresponding additional data to represent the set of the searching criteria and the result.
+
+Vald Metadata includes the user inputted metadata(vector ID) and the vector, and the internal generated UUID.
+
+#### Vald Meta Gateway
+
+The main responsibility of the Vald Meta Gateway is to process the Vald metadata and to forward the information to Vald Backup Gateway.
+
+It will perform the following action:
+
+1. Return error if the user has already input the same vector in Vald
+1. Generate the corresponding UUID for internal use.
+1. Forward the vector ID and UUID request to the Vald Meta.
+1. Forward the vector information (vector ID, vector, and UUID) to Vald Backup Gateway.
+
```

Comment:
why are all numbers 1?

Reply:
It is github markdown coding style.
1. a
1. b
1. c